gubernator/log_parser.py

```diff
@@ -75,15 +75,23 @@ def log_html(lines, matched_lines, hilight_words, skip_fmt):
     return output
 
 
-def digest(data, skip_fmt=lambda l: '... skipping %d lines ...' % l,
-      objref_dict=None, filters=None, error_re=regex.error_re):
+def digest(data, objref_dict=None, filters=None, error_re=regex.error_re,
+    skip_fmt=lambda l: '... skipping %d lines ...' % l, timestamps=None):
+    # pylint: disable=too-many-arguments
     """
     Given a build log, return a chunk of HTML code suitable for
     inclusion in a <pre> tag, with "interesting" errors hilighted.
 
     This is similar to the output of `grep -C4` with an appropriate regex.
     """
     lines = unicode(jinja2.escape(data)).split('\n')
+    if timestamps:
```

Comment:
You should do the combining/sorting outside of digest, and only ever pass a string to digest. Having the type of the arguments depend on a bool is confusing.